### PR TITLE
Picklable

### PR DESCRIPTION
**Pull request by mrocklin**
Supercedes #1457

### CODE REVIEW

**Comment by jcrist** [comment]
This isn't quite correct, as the bag `__init__` adds the `str` accessor. This should be made a property then.

**Comment by mrocklin** [comment]
Whistles dualing banjo tune: 

Fixed.  Also includes tests

File: dask/bag/core.py [hunk 1]
```diff
@@ -316,7 +322,10 @@ def __init__(self, dsk, name, npartitions):
         self.dask = dsk
         self.name = name
         self.npartitions = npartitions
-        self.str = StringAccessor(self)
+
+    @property
+    def str(self):
```
Comment (jcrist) [inline dask/bag/core.py]: Since we're competing, no docstring here for `str` accessor :).

**Comment by jcrist** [comment]
superseded by #1457.

**Pull request closed without merging**